src/move_and_slide.rs

```diff
@@ -99,3 +100,82 @@ pub fn move_and_slide(
         }
     }
 }
+
+fn similar_plane(a: Vec3, b: Vec3) -> bool {
+    a.dot(b) > SIMILARITY_THRESHOLD
+}
```

Comment:
Could be more explicit about names here
```suggestion
fn similar_plane(normal1: Vec3, normal2: Vec3) -> bool {
    normal1.dot(normal2) > SIMILARITY_THRESHOLD
}
```